.github/terraform/main.tf

```diff
@@ -0,0 +1,283 @@
+  name    = "${var.prefix}-firewall"
+  network = google_compute_network.vpc.name
+
+  # Allow SSH, Hazelcast member communication and Hazelcat Management Center website
+  allow {
+    protocol = "tcp"
+    ports    = ["22", "5701-5707", "8080"]
+  }
+
+  allow {
+    protocol = "icmp"
+  }
+}
+
+resource "google_compute_address" "public_ip" {
+  count = var.member_count + 1
+  name  = "${var.prefix}-ip-${count.index}"
+}
+
+############## HAZELCAST MEMBERS #####################
+
+resource "google_compute_instance" "hazelcast_member" {
+  count                     = var.member_count
+  name                      = "${var.prefix}-instance-${count.index}"
+  machine_type              = var.gcp_instance_type
+  allow_stopping_for_update = "true"
+  zone                      = var.zone
+  boot_disk {
+    initialize_params {
+      image = "debian-cloud/debian-9"
+    }
+  }
+
+  labels = {
+    "${var.gcp_label_key}" = var.gcp_label_value
+  }
+
+  network_interface {
+    subnetwork = google_compute_subnetwork.vpc_subnet.self_link
+    access_config {
+      nat_ip = google_compute_address.public_ip[count.index].address
+    }
+  }
+
+  service_account {
+    email  = google_service_account.service_account.email
+    scopes = ["cloud-platform"]
+  }
+
+  metadata = {
+    ssh-keys = "${var.gcp_ssh_user}:${file("${var.local_key_path}/${var.gcp_key_name}.pub")}"
+  }
+
+  connection {
+    host        = self.network_interface[0].access_config[0].nat_ip
+    user        = var.gcp_ssh_user
+    type        = "ssh"
+    private_key = file("${var.local_key_path}/${var.gcp_key_name}")
+    timeout     = "120s"
+    agent       = false
+  }
+  provisioner "remote-exec" {
+    inline = [
+      "mkdir -p /home/${var.gcp_ssh_user}/jars",
+      "mkdir -p /home/${var.gcp_ssh_user}/logs",
+      "sudo apt-get update",
+      "sudo apt-get -y install openjdk-8-jdk wget",
+    ]
+  }
+
+  provisioner "file" {
+    source      = "scripts/start_gcp_hazelcast_member.sh"
+    destination = "/home/${var.gcp_ssh_user}/start_gcp_hazelcast_member.sh"
+  }
+
+  provisioner "file" {
+    source      = "scripts/verify_member_count.sh"
+    destination = "/home/${var.gcp_ssh_user}/verify_member_count.sh"
+  }
+
+  provisioner "file" {
+    source      = "~/lib/hazelcast-gcp.jar"
+    destination = "/home/${var.gcp_ssh_user}/jars/hazelcast-gcp.jar"
+  }
+
+  provisioner "file" {
+    source      = "~/lib/hazelcast.jar"
+    destination = "/home/${var.gcp_ssh_user}/jars/hazelcast.jar"
+  }
+
+  provisioner "file" {
+    source      = "hazelcast.yaml"
+    destination = "/home/${var.gcp_ssh_user}/hazelcast.yaml"
+  }
+
+  provisioner "remote-exec" {
+    inline = [
+      "cd /home/${var.gcp_ssh_user}",
+      "chmod 0755 start_gcp_hazelcast_member.sh",
+      "./start_gcp_hazelcast_member.sh  ${var.gcp_label_key} ${var.gcp_label_value} ",
+      "sleep 5",
+    ]
+  }
+}
+
+resource "null_resource" "verify_members" {
+  count      = var.member_count
```

Comment:
This way we check the logs of each member separately. Why don't we check only one of them - as the membership logs will be the same for both? What I mean is, after verifying the members here:
```
Members {size:2, ver:2} [
  null_resource.verify_members[1] (remote-exec): Member [....]:5701 - 412020b1-b3af-4c0c-b29e-71e38fe2548b this
  null_resource.verify_members[1] (remote-exec): Member [....]:5701 - 3ecf685c-f107-439b-b890-4e8c654e7e8b
]
```
we also check the otherside:
```
Members {size:2, ver:2} [
  null_resource.verify_members[0] (remote-exec): Member [....]:5701 - 412020b1-b3af-4c0c-b29e-71e38fe2548b
  null_resource.verify_members[0] (remote-exec): Member [....]:5701 - 3ecf685c-f107-439b-b890-4e8c654e7e8b this
]
```

Reply:
We could, but checking only one of them wouldn't really create a difference. But if you have a strong opinion about it, I can change it.

Reply:
Not a strong opinion at all. Just getting rid of an extra work. However, making a double check won't be that bad. Let's keep it as it is then.